homeassistant/components/homekit/util.py

```diff
@@ -110,6 +113,59 @@ def validate_media_player_features(state, feature_list):
     return True
 
 
+SpeedRange = namedtuple('SpeedRange', 'start target stop')
+SpeedRange.__doc__ += """: a mapping of descrete Home Assistant speed \
+values to percentage based HomeKit speeds."""
+SpeedRange.start.__doc__ += """Start of the range (inclusive)."""
+SpeedRange.target.__doc__ += """Percentage to use to determine HomeKit \
+percentages from HomeAssistant speed."""
+SpeedRange.stop.__doc__ += """Stop of the range (exclusive)."""
+
+
+class HomeKitSpeedMapping:
+    """Supports conversion between Home Assistant and HomeKit fan speeds."""
+
+    def __init__(self, speed_list):
+        """Initialize a new SpeedMapping object."""
+        if not speed_list or speed_list[0] != fan.SPEED_OFF:
+            _LOGGER.warning("%s does not contain the speed setting "
+                            "%s as its first element. "
+                            "Assuming that %s is equivalent to 'off'.",
+                            speed_list, fan.SPEED_OFF, speed_list[0])
+        speed_ranges = {}
```

Comment:
If we use and `OrderedDict`, we should be able to assum later that the first item is the min speed setting. See `speed_to_states` below.